File: backend/dimensions.py
"""保障维度 —— 单一事实源（权威定义）。

两个语义层：
- `tag`：知识库真实标签名（作战地图按此查文档，必须与 tags 表一致）
- `label`：展示名（保障清单分类名 / UI 显示，可比 tag 更完整）

顺序即此列表顺序。改维度只动这一处。前端对应 frontend/src/dimensions.js（保持一致）。
"""

# (tag, label, icon)
DIMENSION_DEFS = [
    ("事故/故障", "事故/故障", "🚨"),
    ("高可用保障", "高可用保障", "🛡️"),
    ("直播体验", "直播播放体验", "📺"),
    ("成本", "成本", "💰"),
    ("安全", "安全", "🔒"),
    ("业务需求", "业务需求", "🎯"),
]

# 标签名列表（作战地图按标签查文档用）
DIMENSION_TAGS = [d[0] for d in DIMENSION_DEFS]
# 展示名列表（保障清单分类 / 排序用）
DIMENSION_LABELS = [d[1] for d in DIMENSION_DEFS]
# 标签名 → 展示名
TAG_TO_LABEL = {d[0]: d[1] for d in DIMENSION_DEFS}
# 标签名 → 图标
TAG_TO_ICON = {d[0]: d[2] for d in DIMENSION_DEFS}

File: frontend/src/dimensions.js
// 保障维度 —— 前端单一事实源（与后端 backend/dimensions.py 保持一致）。
// tag: 知识库真实标签名；label: 展示名；icon: 图标。顺序即此列表。改维度只动此处。

export const DIMENSION_DEFS = [
  { tag: '事故/故障', label: '事故/故障', icon: '🚨' },
  { tag: '高可用保障', label: '高可用保障', icon: '🛡️' },
  { tag: '直播体验', label: '直播播放体验', icon: '📺' },
  { tag: '成本', label: '成本', icon: '💰' },
  { tag: '安全', label: '安全', icon: '🔒' },
  { tag: '业务需求', label: '业务需求', icon: '🎯' },
]

// 展示名列表（保障清单分类/排序用）
export const DIMENSION_LABELS = DIMENSION_DEFS.map(d => d.label)
// 标签名 → 图标（作战地图按标签名取图标）
export const TAG_TO_ICON = Object.fromEntries(DIMENSION_DEFS.map(d => [d.tag, d.icon]))
